router/gossip.go

```diff
@@ -224,6 +228,16 @@ func (c *GossipChannel) GossipBroadcast(update GossipData) error {
 	return c.relayBroadcast(c.ourself.Name, update)
 }
 
+func (c *GossipChannel) LeaderElect() PeerName {
+	highest := PeerName(0)
+	c.ourself.Router.Peers.ForEach(func(peer *Peer) {
+		if highest < peer.Name {
+			highest = peer.Name
+		}
+	})
+	return highest
+}
+
```

Comment:
Yes, the cast is there to remind me I'm not entirely sure which way to go: 

I think a case can be made that leadership is related to gossip, so it can go on the Gossip interface.  

Or the thing that uses it (Allocator, in the first instance) could be passed a reference to a Leadership interface.  If we go that way, Peers could be the implementer.